tee_supplicant/src/ree_fs.c

```diff
@@ -144,15 +139,11 @@ static int tee_fs_read(size_t num_params, struct tee_param *params)
 		return TEEC_ERROR_ITEM_NOT_FOUND;
 	}
 
-	/*TODO do all safety checks */
-	shm = (struct tee_shm *)params[1].c;
-	len = params[1].b;
-
-	if (!shm || !shm->addr) {
+	buf = tee_param_get_mem(params + 1, NULL);
+	if (!buf) {
 		return TEEC_ERROR_BAD_PARAMETERS;
 	}
-
-	buf = shm->addr;
+	len = params[1].b;
```

Comment:
As you are touching this part of the code anyways...

You already have `MEMREF_SIZE`, why you are not using it there?